### ISSUE TITLE
mt7603 on a Xiaomi 3g

### ISSUE THREAD

**Issue by tkso1997** [Nossiac/mtk-openwrt-feeds#93]
Is there any news for mt7603 on a Xiaomi 3g? With openwrt drivers the 2.4ghz 7603 is basically not usable. the 5ghz works pretty good! Can we mix both drivers, for instance mtk7603 for 2.4ghz and the opensource mt7012 from openwrt  for 5ghz?

**Comment by nicefile** [comment]
Yes we can  . Take look here for my custom 4.14.66 build  with opensource 5Ghz and close source 2.4: https://drive.google.com/drive/folders/1_WRgzpJyFr3GqQvtHelz2i_4nhI9KMyB

**Comment by tkso1997** [comment]
Cool! 
Is there anything I need to be prepared when installing it? Can I just do a sysupgrade or do I have to flash the partitions directly?
Are you using it on your router, how is it working?

Thanks 

Thorsten

**Comment by nicefile** [comment]
I've did flash from breed to "OpenWRT-v18.06.1-MiR3G-ForBreed_Full.bin" . now included in g drive . Then sysupgrade to my firmware (pass is 1234567890 afair). No intensive testing done with my build but it worked with few devices connected and 2.4 Ghz was fast as it should .

**Comment by nicefile** [comment]
https://drive.google.com/drive/folders/10cJSdaFMHwEjphrtaMpVfhQoO1AqksxI  same with breed flashed openwrt

**Comment by julianocss** [comment]
Hello @nicefile.

Can you tell me how you mixed 2.4 and 5 ghz drivers?
Another doubt: This repo is a kernel dependent? I cannot clone him and build a new firmware using latest openwrt snapshot?

Thanks.

**Comment by julianocss** [comment]
Hello @brainfog0.

I made the first two steps, the firmware was built without errors. 
But i am afraid to flash and brick the router. 

Can you help-me in these steps?
"added script to fetch eeprom from flash, and added module to /etc/modules.d for autobooting."

Thanks. 
Juliano

**Comment by nicefile** [comment]
@julianocss 
easy: https://git.openwrt.org/?p=openwrt/openwrt.git;a=shortlog;h=refs/heads/openwrt-18.06;pg=1
look for commit just before update to 4.14.67 and that will be "kernel: bump 4.9 to 4.9.124" click tree next to it then snapshot  to download tgz

**Comment by julianocss** [comment]
Thanks @nicefile, it works.

And what is these steps that @brainfog0 said: "added script to fetch eeprom from flash, and added module to /etc/modules.d for autobooting." 
My router is an Archer C50 V3 (MT7628AN + MT7612E). Can you help-me how to mix the drivers?

Juliano

**Comment by nicefile** [comment]
@julianocss 
Additional script is not needed .  You need extract 512b eeprom from 'factory' (mtd3 in my case) like 
dd if=/dev/mtd3 of=/etc/wireless/mt7603/mt7603.eeprom.bin bs=1 count=512 .Should work from start . Try build Openwrt with nossiac modules first to see if they work . Then if not  post the issue  in new thread and we will see .

**Comment by brainfog0** [comment]
@julianocss
sorry, was no clear enough. i meant adding this command mentioned by nicefile above to uci-defaults, so you don't need to use it manually after installing firmware.
autobooting not needed for tests, i just made it later for convenience. first you want try to load driver manually with "insmod mt7603" command.
if you have doubts about bricking router, try to find usb uart adapter or flash breed u-boot to your router (be careful with flashing u-boot though)

**Comment by julianocss** [comment]
Hello @nicefile and @brainfog0.

So, the steps will be:
1) Build a firmware with 4.14.66 kernel without mt76 drivers.
2) Download the file " mt7612-for-mt76x8-linux-4.14.66.ko "  (this is the correct file for my router?).
3) Install manually using isnmod ( the command is only isnmod 'downloaded_file.ko' )?

Thanks.
Juliano

**Comment by brainfog0** [comment]
@julianocss
instead of downloading manually would be easier to add feeds from this repository (look readme.md for feeds instructions).
then enable needed modules in menuconfig and enable plugin for luci for easier configuration, and build firmware.
yes, to load module into a kernel, use command insmod 'module_name'

**Comment by julianocss** [comment]
Ok, thanks @brainfog0 .

Do you know which modules will be necessary for my router?
Just mt7612? Or mt7612 + mt7628 ?

Regards.
Juliano

**Comment by brainfog0** [comment]
you may try both. on xiaomi r3g mt7603 worked fine, but mt7612 crashed router.
maybe on your hardware it will work better

**Comment by julianocss** [comment]
@brainfog0 

Both radios (2.4 and 5 Ghz) works with only MT7603 driver?

Juliano

**Comment by brainfog0** [comment]
@julianocss 
nope. mt7603 - 2.4 GHz module, mt7612 - 5 GHz module

**Comment by julianocss** [comment]
Hello @nicefile and @brainfog0 

I tried to compile the firmware with 4.14.66 snapshot build, but image builder do not completed and gave a lot of erros. I think that could be a problem with feeds, like luci-app-adblock, luci-app-sqm, etc.
Do you have any suggestions? I thought about creating an image of stable version 18.06.1, but the Kernel version is different...

Juliano

**Comment by julianocss** [comment]
Hello @nicefile and @brainfog0

I was able to compile the firmare.
The driver did not load automatically. After you manually load it and start the interface, the router restarts.

Do you have any idea what would be the problem?

I use this Openwrt commit:

[https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=e3022727658166e736198529582a46abf2397ea4](url)

Regards,
Juliano

**Comment by tkso1997** [comment]
Hello @nicefile 
> https://drive.google.com/drive/folders/10cJSdaFMHwEjphrtaMpVfhQoO1AqksxI same with breed flashed openwrt

Is it working okay for you and could you share it again (link is down)?

One other thing: How did you manage to prepare the firmware that it is flashable with breed, would be very interested in it! What are the commands to turn a openwrt image into a breed compatible file?

Regards
Thorsten

**Comment by tkso1997** [comment]
Hella @nicefile @all,

I compiled OpenWrt with kernel-4.14.93 your newest 4.14.93-mt7603 and the open source 5GHz mt76 driver. So far it works quite nice. Any tips on how to make the 2.4GHz a bit more responsive? Throughput is okay but its a bit laggy. Any recommendations are very welcome.

So far the combination of your closed-source 2.4GHz driver plus the open source 5GHz one seems to be the best for Xiaomi 3g. 

If someone wants to try my version feel free:

[https://www.dropbox.com/s/ehbnvj94z5ecb9w/openwrt-ramips-mt7621-mir3g-squashfs-sysupgrade.bin?dl=0](url)

**Comment by nicefile** [comment]
Hello @ALL 
kernel 4.14.93 from me  with mt76 from 31.01 and shed_cake 15.01
I've added script  and firmware do flash it under breed :
https://drive.google.com/drive/folders/1P9tsRaSeHpEyqMZBZ0HwNzIhYJgb6OxG?usp=sharing

**Comment by pataha** [comment]
@tkso1997 can I flash your firmware through luci upgrade page?

**Comment by tkso1997** [comment]
@pataha yes you can

**Comment by nicefile** [comment]
@ALL yet another update with mt version 2019-03-27 and 4.14.93
https://drive.google.com/open?id=1eYKT40xQY4m-wt-TaExBR_bL7-7gBdsO
It supposed to have 1100Mhz overclock (patch based ) but cannot confirm that on my device . Just don't see it

**Comment by tr4usut** [comment]
Hope you can share build usually when have new build. tks you @nicefile :)

**Comment by jinglei207** [comment]
> 
> 
> @ALL yet another update with mt version 2019-03-27 and 4.14.93
> https://drive.google.com/open?id=1eYKT40xQY4m-wt-TaExBR_bL7-7gBdsO
> It supposed to have 1100Mhz overclock (patch based ) but cannot confirm that on my device . Just don't see it

Hi how did you add the 5ghz driver? with `ifconfig -a`, i saw that the interface name is `rai0`, but under luci, it shows as `ra0`, i can't enable it.
https://github.com/Nossiac/mtk-openwrt-feeds/issues/116

**Comment by 9000000** [comment]
> @ALL yet another update with mt version 2019-03-27 and 4.14.93
> https://drive.google.com/open?id=1eYKT40xQY4m-wt-TaExBR_bL7-7gBdsO
> It supposed to have 1100Mhz overclock (patch based ) but cannot confirm that on my device . Just don't see it

I cannot download it

**Comment by nicefile** [comment]
> > @ALL yet another update with mt version 2019-03-27 and 4.14.93
> > https://drive.google.com/open?id=1eYKT40xQY4m-wt-TaExBR_bL7-7gBdsO
> > It supposed to have 1100Mhz overclock (patch based ) but cannot confirm that on my device . Just don't see it
> 
> I cannot download it

Sorry. I've trashed this project  .I'ts possible to build it manually but that version will be outdated .

**Comment by 9000000** [comment]
> > > @ALL yet another update with mt version 2019-03-27 and 4.14.93
> > > https://drive.google.com/open?id=1eYKT40xQY4m-wt-TaExBR_bL7-7gBdsO
> > > It supposed to have 1100Mhz overclock (patch based ) but cannot confirm that on my device . Just don't see it
> > 
> > 
> > I cannot download it
> 
> Sorry. I've trashed this project .I'ts possible to build it manually but that version will be outdated .

Thank you I have found some similar new projects but there is so much unnecessary software for me, I just need to be simple.
https://github.com/coolsnowwolf/lede
https://github.com/Lienol/openwrt
